tee_client_api: Change type of shared memory id parameter

oleksiimoisieiev opened the pull request:
Switch SharedMemory id parameter from int to uint64_t. In original optee-client library for Linux int is enough because fd is stored there. But in Zephyr id (which is shm_ref for OP-TEE) is a pointer to shm structure so generic `int` type which is 32-bit in Zephyr is not enough to store structure pointer and may lead to passing different highbits of shm_ref on SHM_REGISTER and INVOKE_COMMAND calls.

lorc commented:
Minor corrections: 
- `fd` is not a parameter, but field in `TEEC_SharedMemory` structure.
- Linux uses not `fd`, it is just sequence of natural numbers: 1, 2, 3, 4, ...

Have you run whole test suite, BTW? I am asking because there are places where `-1` is assigned to `id` field. I believe we need to replace those `-1`'s with something like `TEEC_SHM_INVALID_ID` with value of `~0`.

oleksiimoisieiev commented:
> Minor corrections:
> 
>     * `fd` is not a parameter, but field in `TEEC_SharedMemory` structure.
> 
>     * Linux uses not `fd`, it is just sequence of natural numbers: 1, 2, 3, 4, ...
> 
> 
> Have you run whole test suite, BTW? I am asking because there are places where `-1` is assigned to `id` field. I believe we need to replace those `-1`'s with something like `TEEC_SHM_INVALID_ID` with value of `~0`.

Fixed according to the comments. Also - yes I have checked on h3ulcb and on rcar_spyder. Everything seems to be fine. It is unlikely to have pointer but I've added `TEEC_SHM_INVALID_ID`.

lorc reviewed (approved):
Thanks

`Reviewed-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>`
